xonsh/history.py

```diff
@@ -301,28 +287,20 @@ def _curr_session_parser(hist=None, **kwargs):
     """
     if hist is None:
         hist = builtins.__xonsh_history__
-    if not hist:
-        return None
-    start_times = [start for start, end in hist.tss]
-    names = [name[:-1] if name.endswith('\n') else name
-             for name in hist.inps]
-    commands = enumerate(zip(names, start_times))
-    return [(c, t, ind) for ind, (c, t) in commands]
+    start_times = (start for start, end in hist.tss)
+    names = (name.rstrip() for name in hist.inps)
+    for ind, (c, t) in enumerate(zip(names, start_times)):
+        yield (c, t, ind)
 
 
 def _zsh_hist_parser(location=None, **kwargs):
-    default_location = os.path.join('~', '.zsh_history')
-    location_list = [os.path.join('~', '.zshrc'),
-                     os.path.join('~', '.zprofile')]
+    """Yield commands from zsh history file"""
     if location is None:
-        location = _find_histfile_var(location_list, default_location)
-    z_hist_formatted = []
-    if location and os.path.isfile(location):
-        with open(location, 'r', errors='backslashreplace') as z_file:
-            z_txt = z_file.read()
-            z_hist = z_txt.splitlines()
-            if z_hist:
-                for ind, line in enumerate(z_hist):
+        location = _find_histfile_var(['~/.zshrc', '~/.zprofile'], '~/.zsh_history')
```

Comment:
oh yea that's any issue, i didn't change the paths, no idea where zsh saves rc and stuff on windows. will look that up.